src/chromo/common.py

```diff
@@ -543,6 +537,43 @@ def __getnewargs__(self):
         # upon unpickling, create EventData object instead of MCEvent object
         return (EventData,)
 
+    def _add_init_beam_info(self):
+        pass
+
+    def _fill_initial_beam(self):
+        beam = self.kin._get_beam_data(self._generator_frame)
+        for field, beam_field in beam.items():
+            event_field = getattr(self, field)
+            if event_field is not None:
+                if field == "mothers":
+                    # all mothers that looks like [-1, -1], [0, -1], [1, -1], [1, 1]
```

Comment:
It was not correct. Indeed, spectators should only attach to only one of the mothers. However, it turned out that the DpmJet and Phojet history is too complex for hepmc. Because of that the history produced by these models is left as it is. Only indicies in mothers and daughters are shifted to satisfy zero indexing.